### PR TITLE
Flav/add user to access key

### LINKED ISSUE #3075: Record the user who create the API Keys

### PR DESCRIPTION
**Pull request by flvndvd**
This PR resolves https://github.com/dust-tt/dust/issues/3075 by introducing a new column to the keys table that records the creator of each key. Additionally, the user interface has been updated to show both the creator's name and the creation date. For existing keys where the creator information is not available, the displayed author will be listed as "unknown."

image

### CODE REVIEW

File: front/pages/w/[wId]/a/index.tsx [hunk 1]
```diff
@@ -141,53 +141,61 @@ export function APIKeys({ owner }: { owner: WorkspaceType }) {
...
+                              ? "bg-green-100 text-green-800"
+                              : "ml-6 bg-gray-100 text-gray-800"
+                          )}
+                        >
+                          {key.status === "active" ? "active" : "revoked"}
+                        </p>
+                      </div>
+                    </div>
+                    <p className="front-normal text-xs text-element-700">
+                      Created by {key.creator ?? "unknown"}{" "}
```
Comment (spolu) [inline front/pages/w/[wId]/a/index.tsx:197]: Let's do just `Created timeAgo` when key.creator is not set ?

**Review by spolu — APPROVED** [review]
LGTM! left 2 small comments but feel free to merge :100:

File: types/src/front/user.ts [hunk 2]
```diff
@@ -31,3 +31,7 @@ export type UserMetadataType = {
   key: string;
   value: string;
 };
+
+export function formatUserFullName(user: {firstName?: string, lastName?: string | null}) {
```
Comment (spolu) [inline types/src/front/user.ts:35]: typing errors

File: front/pages/api/w/[wId]/keys/index.ts [hunk 3]
```diff
@@ -38,19 +39,31 @@ async function handler(
...
+          },
+        ],
       });
 
       res.status(200).json({
         keys: keys.map((k) => {
           return {
+            createdAt: k.createdAt.getTime(),
+            creator: formatUserFullName(k.user),
+            isSystem: k.isSystem,
```
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/pages/api/w/[wId]/keys/index.ts]: Removed.

File: front/pages/w/[wId]/a/index.tsx (hunk 1 above)
Comment (flvndvd) [reply front/pages/w/[wId]/a/index.tsx]: Done.

**Pull request merged**